# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by scripts/Create-CopyPatch.
# 
# T2 SDE: package/*/cmake/uname.patch
# Copyright (C) 2006 - 2021 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Cmake combines stderr and stdout, as T2 emmits a warning to stderr
we need to get rid of it.

NOTE: Don't remove the | cat, as cmake appends 2>&1 so it would undo it!

  - Rene Rebe <rene@exactcode.de>

--- cmake-3.21.2/Modules/CMakeDetermineSystem.cmake.vanilla	2021-08-29 19:02:39.382626212 +0200
+++ cmake-3.21.2/Modules/CMakeDetermineSystem.cmake	2021-08-29 19:04:58.716633400 +0200
@@ -42,10 +42,10 @@
       unset(_CMAKE_HOST_SYSTEM_MAJOR_VERSION)
       unset(_CMAKE_HOST_SYSTEM_MINOR_VERSION)
     else()
-      exec_program(${CMAKE_UNAME} ARGS -r OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_VERSION)
+      exec_program(${CMAKE_UNAME} ARGS -r 2>/dev/null | cat OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_VERSION)
     endif()
     if(CMAKE_HOST_SYSTEM_NAME MATCHES "Linux|CYGWIN.*|MSYS.*|^GNU$|Android")
-      exec_program(${CMAKE_UNAME} ARGS -m OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_PROCESSOR
+      exec_program(${CMAKE_UNAME} ARGS -m 2>/dev/null | cat OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_PROCESSOR
         RETURN_VALUE val)
     elseif(CMAKE_HOST_SYSTEM_NAME MATCHES "Darwin")
       # If we are running on Apple Silicon, honor CMAKE_APPLE_SILICON_PROCESSOR.
@@ -55,10 +55,10 @@
       exec_program(arch ARGS -s OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_PROCESSOR
         RETURN_VALUE val)
     else()
-      exec_program(${CMAKE_UNAME} ARGS -p OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_PROCESSOR
+      exec_program(${CMAKE_UNAME} ARGS -p 2>/dev/null | cat OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_PROCESSOR
         RETURN_VALUE val)
       if("${val}" GREATER 0)
-        exec_program(${CMAKE_UNAME} ARGS -m OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_PROCESSOR
+        exec_program(${CMAKE_UNAME} ARGS -m 2>/dev/null | cat OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_PROCESSOR
           RETURN_VALUE val)
       endif()
     endif()
--- cmake-3.20.0/Tests/RunCMake/ParseImplicitData/CMakeLists.txt.vanilla	2021-04-06 14:24:14.646308256 +0200
+++ cmake-3.20.0/Tests/RunCMake/ParseImplicitData/CMakeLists.txt	2021-04-06 14:25:00.870310641 +0200
@@ -21,7 +21,7 @@
 
 project(gen_implicit_include_data ${LANGUAGES})
 
-set(UNAME "${CMAKE_SYSTEM_NAME}" CACHE STRING "System uname")
+set(UNAME "${CMAKE_SYSTEM_NAME}" 2>/dev/null CACHE STRING "System uname")
 string(TOLOWER "${UNAME}" UNAME)
 message("Generate input for system type: ${UNAME}")
 
